src/List.js

```diff
@@ -57,7 +57,7 @@ export class List extends IndexedCollection {
 
   get(index, notSetValue) {
     index = wrapIndex(this, index);
-    if (index < 0 || index >= this.size) {
+    if (index < 0 || index >= this.size || index !== index) {
```

Comment:
Rather than doing this, we can flip the if check and only do the get if it's safe, rather than short-circuiting.